app/models/solidus_paypal_commerce_platform/gateway.rb

```diff
@@ -70,6 +70,23 @@ def void(_response_code, options)
       @client.execute_with_response(request)
     end
 
+    # Will void the payment depending on its state or return false
+    #
+    # If the payment has not yet been captured, we can void the transaction.
+    # Otherwise, we return false so Solidus creates a refund instead.
+    #
+    # https://developer.paypal.com/docs/api/payments/v2/#authorizations_void
+    #
+    # @api public
+    # @param payment [Spree::Payment] the payment to void
+    # @return [Response|FalseClass]
+    def try_void(payment)
```

Comment:
This should be implemented on the payment method. 

See the implementation in `solidus_core`:

1. https://github.com/solidusio/solidus/blob/master/core/app/models/spree/payment/cancellation.rb#L29
2. https://github.com/solidusio/solidus/blob/master/core/app/models/spree/payment_method.rb#L169-L186

Reply:
I agree with @tvdeyen, and taking it a step further, I'm wondering if this should be the default implementation of `Spree::Payment#try_void`, thoughts?

Reply:
Yes, this could work as a default implementation.